app/controllers/rails/view_components_controller.rb

```diff
@@ -16,23 +16,31 @@ class Rails::ViewComponentsController < Rails::ApplicationController # :nodoc:
   def index
     @previews = ViewComponent::Preview.all
     @page_title = "Component Previews"
+    opts = default_preview_layout.nil? ? {} : { layout: default_preview_layout }
+    prepend_application_view_paths if opts.key?(:layout)
```

Comment:
Can we set the layout explicitly rather than the indirection of `prepend_application_view_paths`.
```suggestion
    opts = default_preview_layout.nil? ? {} : { layout: Rails.root.join("app/views/layouts",  default_preview_layout) }
```

Reply:
@jonspalmer Well prepending the app's view path allows the layout to use partials from the app (shared/head, etc.), so that needs to be preserved.

Reply:
Got it. And I'm guessing that sadly resolving those partial's isn't local to the layout that is picked. It relies of view paths?

Reply:
Let me see if I can refactor the layout hash stuff into a separate method so the individual actions in this controller are simpler.